nanoFirmwareFlasher.Library/FirmwareArchiveManager.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace nanoFramework.Tools.FirmwareFlasher
+{
+    /// <summary>
+    /// The manager of a firmware archive directory. By default, all firmware is downloaded from the online
+    /// CloudSmith repository or obtained from a global cache. In general the latest version is used. As a result,
+    /// subsequent executions of the nanoFirmwareFlasher may yield different versions of the firmware.
```

Comment:
Again, no need to add the motivation in the class description. Please stick to the explanation on what the class does/is used for.

Reply:
Ok.